Utils/azuretools-core/src/com/microsoft/azuretools/sdkmanage/AzureCliAzureManager.java

```diff
@@ -0,0 +1,236 @@
+import com.microsoft.azure.management.resources.Subscription;
+import com.microsoft.azure.management.resources.Tenant;
+import com.microsoft.azuretools.adauth.PromptBehavior;
+import com.microsoft.azuretools.authmanage.AuthMethod;
+import com.microsoft.azuretools.authmanage.AzureManagerFactory;
+import com.microsoft.azuretools.authmanage.CommonSettings;
+import com.microsoft.azuretools.authmanage.Environment;
+import com.microsoft.azuretools.authmanage.SubscriptionManager;
+import com.microsoft.azuretools.authmanage.models.AuthMethodDetails;
+import com.microsoft.azuretools.azurecommons.helpers.Nullable;
+import com.microsoft.azuretools.telemetry.TelemetryInterceptor;
+import com.microsoft.azuretools.utils.AzureRegisterProviderNamespaces;
+import com.microsoft.azuretools.utils.Pair;
+import com.microsoft.rest.credentials.ServiceClientCredentials;
+import org.apache.commons.lang3.StringUtils;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+import static com.microsoft.azuretools.Constants.FILE_NAME_SUBSCRIPTIONS_DETAILS_AZ;
+import static com.microsoft.azuretools.authmanage.Environment.ENVIRONMENT_LIST;
+
+public class AzureCliAzureManager extends AzureManagerBase {
+    private static final String FAILED_TO_AUTH_WITH_AZURE_CLI = "Failed to auth with Azure CLI";
+    private static final String UNABLE_TO_GET_AZURE_CLI_CREDENTIALS = "Unable to get Azure CLI credentials, " +
+            "please ensure you have installed Azure CLI and signed in.";
+
+    private static Settings settings;
+    private Azure.Authenticated authenticated;
+    private AzureCliCredentials azureCliCredentials;
+    private SubscriptionManager subscriptionManager;
+
+    static {
+        settings = new Settings();
+        settings.setSubscriptionsDetailsFileName(FILE_NAME_SUBSCRIPTIONS_DETAILS_AZ);
+    }
+
+    private static class LazyLoader {
+        static final AzureCliAzureManager INSTANCE = new AzureCliAzureManager();
+    }
+
+    public static class AzureCliAzureManagerFactory implements AzureManagerFactory {
+
+        @Override
+        public @Nullable AzureManager factory(AuthMethodDetails authMethodDetails) {
+            return getInstance().isSignedIn() ? getInstance() : null;
+        }
+
+        @Override
+        public AuthMethodDetails restore(final AuthMethodDetails authMethodDetails) {
+            try {
+                getInstance().signIn();
+            } catch (AzureExecutionException e) {
+                // Catch the exception when restore
+            } finally {
+                return authMethodDetails;
+            }
+        }
+    }
+
+    public static AzureCliAzureManager getInstance() {
+        return LazyLoader.INSTANCE;
+    }
+
+    @Override
+    public Azure getAzure(String sid) throws IOException {
+        if (sidToAzureMap.containsKey(sid)) {
+            return sidToAzureMap.get(sid);
+        }
+        final Azure azure = auth().withSubscription(sid);
```

Comment:
What will happen if the sid is not found?

Reply:
Azure SDK should throw exception, but users could only select subscription from list provided by us, which was get from Azure SDK , so this should not be a case